src/app/app.component.ts

```diff
@@ -268,13 +348,74 @@ export class AppComponent implements AfterViewInit {
     this.entrySelected = treeNodesSelected?.length > 0 ? treeNodesSelected[0] : undefined;
   }
 
+  async onToolbarOptionSelected(event) {
+    const customEvent = event as CustomEvent<ToolbarOption>;
+    const optionSelected: ToolbarOption = customEvent.detail;
+    await optionSelected.tag.handler();
+  }
+
+  openNewFolderDialog(): void {
+    this.dialog.open(NewFolderModalComponent, {
+      data: { makeNewFolder: this.makeNewFolder.bind(this) },
+    });
+  }
+
+  openEditColumnsDialog(): void {
+    this.dialog.open(EditColumnsModalComponent, {
+      data: { columnsSelected: this.selectedColumns, allColumnOptions: this.allPossibleColumns, updateColumns: this.updateColumns.bind(this),},
+    });
+  }
+
+  async makeNewFolder(folderName: string) {
+    if (folderName) {
+      if (!this.lfRepositoryBrowser?.nativeElement?.currentFolder) {
+        throw new Error('repositoryBrowser has no currently opened folder.');
```

Comment:
These should be "human-readable" sentences since they will be caught and displayed in UI.
In a production environment these should be localized